From 58e124e9f24435a710b396ea58806647f74a4deb Mon Sep 17 00:00:00 2001
From: MrIvanPlays <pekov.ivan@abv.bg>
Date: Sun, 28 Apr 2019 09:53:09 +0300
Subject: [PATCH] Remove entity rewrite

This patch gets rid of the entity rewrite.
Suggesting to NOT change the disable_entity_metadata_rewrite
option into waterfall.yml. If you change it back to false, this will
force the proxy to send packets which are useless, that can cause
memory leaks or more cpu usage

diff --git a/proxy/src/main/java/io/github/waterfallmc/waterfall/conf/WaterfallConfiguration.java b/proxy/src/main/java/io/github/waterfallmc/waterfall/conf/WaterfallConfiguration.java
index f316810b..a5ec787c 100644
--- a/proxy/src/main/java/io/github/waterfallmc/waterfall/conf/WaterfallConfiguration.java
+++ b/proxy/src/main/java/io/github/waterfallmc/waterfall/conf/WaterfallConfiguration.java
@@ -53,7 +53,7 @@ public abstract class WaterfallConfiguration extends Configuration { // IvanCord
      */
     private boolean allowEmptyPackets = false;
 
-    private boolean disableEntityMetadataRewrite = false;
+    private boolean disableEntityMetadataRewrite = true; // IvanCord - default to true
 
     @Override
     public void load() {
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
index 4739050e..a8b8ba8b 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
@@ -136,7 +136,9 @@ public class DownstreamBridge extends PacketHandler
     @Override
     public void handle(PacketWrapper packet) throws Exception
     {
+        /* IvanCord - get rid of entity rewrites
         con.getEntityRewrite().rewriteClientbound( packet.buf, con.getServerEntityId(), con.getClientEntityId(), con.getPendingConnection().getVersion() );
+         */
         con.sendPacket( packet );
     }
 
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
index 47fe9710..2f880268 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java
@@ -118,7 +118,9 @@ public class UpstreamBridge extends PacketHandler
     {
         if ( con.getServer() != null )
         {
+            /* IvanCord - get rid of entity rewrites
             con.getEntityRewrite().rewriteServerbound( packet.buf, con.getClientEntityId(), con.getServerEntityId(), con.getPendingConnection().getVersion() );
+             */
             con.getServer().getCh().write( packet );
         }
     }
-- 
2.21.0.windows.1

